myia/parser.py

```diff
@@ -284,6 +322,38 @@ def process_NameConstant(self, block: 'Block',
         """Process special constants: `True`, `False`, `None`, etc."""
         return self.environment.map(node.value)
 
+    def process_Tuple(self, block: 'Block', node: ast.Tuple) -> ANFNode:
+        """Process tuple literals."""
+        op = Constant(primops.cons_tuple)
+        elts = [self.process_node(block, e) for e in node.elts]
+
+        def cons(elts):
+            if len(elts) == 0:
+                return Constant(())
+            else:
+                x, *rest = elts
+                return Apply([op, x, cons(rest)], block.graph)
+        return cons(elts)
+
+    def process_Subscript(self, block: 'Block',
+                          node: ast.Subscript) -> ANFNode:
+        """Process subscripts: `x[y]`."""
+        op = self.environment.map(operator.getitem)
+        value = self.process_node(block, node.value)
+        slice = self.process_node(block, node.slice)
+        return Apply([op, value, slice], block.graph)
+
+    def process_Index(self, block: 'Block', node: ast.Index) -> ANFNode:
+        """Process subscripts with simple index: `x[y]`."""
```

Comment:
This docstring is really confusing for someone not familiar with the python AST.  Maybe something like: `Process subscript indexes`